### PR TITLE
[TextInput] Revert to previous value on esc key press

### PR DESCRIPTION
**Pull request by ellthompson**
Fixes https://github.com/playcanvas/editor/issues/207

### CODE REVIEW

File: src/components/TextInput/index.js [hunk 1]
```diff
@@ -105,6 +105,7 @@ class TextInput extends Element {
     _onInputFocus(evt) {
         this.class.add(pcuiClass.FOCUS);
         this.emit('focus', evt);
+        this._prevValue = this.value;
```
Comment (slimbuck) [inline src/components/TextInput/index.js:108]: might want to declare/create this member in the constructor?

File: src/components/TextInput/index.js [hunk 2]
```diff
@@ -113,7 +114,10 @@ class TextInput extends Element {
     }
 
     _onInputKeyDown(evt) {
-        if ((evt.keyCode === 27 && this.blurOnEscape) || (evt.keyCode === 13 && this.blurOnEnter)) {
+        if (evt.keyCode === 27 && this.blurOnEscape) {
+            this.value = this._prevValue;
```
Comment (vkalpias) [inline src/components/TextInput/index.js:118]: Won't setting 'value' result in a value recorded when the value has not changed yet? Maybe the value should be changed on the input field directly instead with no 'change' events fired?

Comment (ellthompson) [reply src/components/TextInput/index.js:118]: Yep good shout, i'll switch it over.

File: src/components/TextInput/index.js (hunk 1 above)
Comment (ellthompson) [reply src/components/TextInput/index.js:108]: Will do 👍

**Review by vkalpias — APPROVED** [review]

**Pull request merged**